src/Type/Constant/ConstantArrayType.php

```diff
@@ -314,7 +315,13 @@ public function equals(Type $type): bool
 
 	public function isCallable(): TrinaryLogic
 	{
-		$typeAndMethod = $this->findTypeAndMethodName();
+		// @todo is there a way to fetch the full scope here.
+		// An OutOfClassScope causes frivolous errors when checking if a
+		// non-static method was invoked statically in its own class, which
+		// is allowed per: https://3v4l.org/mTa9S.
+		// Should add an optional scope parameter to isCallable and if null,
+		// create a new OutOfClassScope instance.
+		$typeAndMethod = $this->findTypeAndMethodName(new OutOfClassScope());
```

Comment:
After circling around this issue for the past few weeks, I think we need to pass the scope to `isCallable`. Knowing if something is callable depends on the scope so we can check the current class. A non-static method on a class can be invoked statically if it is within the same class. This is will always be false given `OutOfClassScope`.

This would also improve detection in `ConstantStringType`. I didn't want to make the big change without first discussing.

Reply:
The `OutOfClassScope` scope here is what is causing the test failures for `CallStaticMethodsRuleTest::testBug1971`.

```
1) PHPStan\Rules\Methods\CallStaticMethodsRuleTest::testBug1971
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'16: Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{class-string<static(Bug1971\HelloWorld)>, 'sayHello2'} given.
+'14: Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{'Bug1971\\HelloWorld', 'sayHello'} given.
+15: Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{class-string<static(Bug1971\HelloWorld)>, 'sayHello'} given.
+16: Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{class-string<static(Bug1971\HelloWorld)>, 'sayHello2'} given.
 '
```

Reply:
What about a new class called `InAnyClassScope` that returns `true` from all the `can*` methods?

We'd still get the actual validation because `getCallableParametersAcceptors` accepts the real scope.